apps/src/code-studio/hiddenStageRedux.js

```diff
@@ -5,57 +5,64 @@
 import $ from 'jquery';
 import { INIT_PROGRESS } from './progressRedux';
 import experiments from '@cdo/apps/experiments';
+import Immutable from 'immutable';
 
-export const UPDATE_HIDDEN_STAGES = 'hiddenStage/UPDATE_HIDDEN_STAGES';
+export const UPDATE_HIDDEN_STAGE = 'hiddenStage/UPDATE_HIDDEN_STAGE';
 export const ALLOW_HIDEABLE = 'hiddenStage/ALLOW_HIDEABLE';
 
+const STUDENT_SECTION_ID = 'STUDENT';
+
 export const hiddenStagesEnabled = () => experiments.isEnabled('hiddenStages');
 
-// State is a key indicating whether or not we've received hidden stage info
-// from the server, and then a set of stageIds mapped to bools that are true
-// if that stage is hidden
-const initialState = {
-  initialized: false
-};
+const initialState = Immutable.fromJS({
```

Comment:
Converted this subsection of our redux store to use Immutable while I was changing things, mostly because I wanted to be able to easily change values a few levels deep (and Immutable provides a nice API for this with `setIn`)